e2e/suite_test.go

```diff
@@ -19,67 +19,64 @@ import (
 	"context"
 	"fmt"
 	"os"
-	"strconv"
 	"testing"
 	"time"
 
-	"github.com/compose-spec/compose-go/cli"
 	"github.com/onsi/ginkgo/v2"
 	"github.com/onsi/gomega"
 	"github.com/sirupsen/logrus"
 
 	"github.com/openclarity/vmclarity/e2e/testenv"
+	"github.com/openclarity/vmclarity/e2e/testenv/types"
 	"github.com/openclarity/vmclarity/pkg/shared/backendclient"
 	"github.com/openclarity/vmclarity/pkg/shared/log"
 )
 
 var (
-	testEnv *testenv.Environment
-	client  *backendclient.BackendClient
+	testEnv  types.Environment
+	client   *backendclient.BackendClient
+	reuseEnv bool
 )
 
-func TestIntegrationTest(t *testing.T) {
+func TestEndToEnd(t *testing.T) {
 	gomega.RegisterFailHandler(ginkgo.Fail)
-	ginkgo.RunSpecs(t, "Run integration tests")
+	ginkgo.RunSpecs(t, "Run end-to-end tests")
 }
 
 func beforeSuite(ctx context.Context) {
 	var err error
 
-	ginkgo.By("creating test environment")
+	ginkgo.By("initializing test environment")
 	log.InitLogger(logrus.DebugLevel.String(), os.Stderr)
 	logger := logrus.WithContext(ctx)
 	ctx = log.SetLoggerForContext(ctx, logger)
 
-	opts, err := cli.NewProjectOptions(
-		[]string{"../installation/docker/docker-compose.yml", "docker-compose.override.yml"},
-		cli.WithName("vmclarity-e2e"),
-		cli.WithResolvedPaths(true),
-	)
+	config, err := testenv.NewConfig()
 	gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
-	err = cli.WithOsEnv(opts)
-	gomega.Expect(err).NotTo(gomega.HaveOccurred())
-
-	var reuseEnv bool
-	if reuseEnv, _ = strconv.ParseBool(os.Getenv("USE_EXISTING")); reuseEnv {
-		logger.Info("reusing existing environment...", "use_existing", reuseEnv)
-	}
+	reuseEnv = config.Reuse
 
-	testEnv, err = testenv.New(opts, reuseEnv)
+	testEnv, err = testenv.New(config)
 	gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
-	ginkgo.By("starting test environment")
-	err = testEnv.Start(ctx)
-	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	if !reuseEnv {
+		ginkgo.By("setup test environment")
+		err = testEnv.SetUp(ctx)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
+		ginkgo.By("starting test environment")
+		err = testEnv.Start(ctx)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	}
```

Comment:
We could log "reusing existing environment" when `reuseEnv` true

Reply:
and we can use `config.Reuse` directly here

Reply:
We would need to store config instead. But will make the change.

Reply:
Both fixed.